swupd/config.go

```diff
@@ -18,56 +18,88 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+	"path/filepath"
 	"strconv"
 	"strings"
 
 	"github.com/go-ini/ini"
 )
 
-var emptyDir = "/var/lib/update/empty/"
-var imageBase = "/var/lib/update/image/"
-var outputDir = "/var/lib/update/www/"
-var debuginfoBanned = true
-var debuginfoLib = "/usr/lib/debug"
-var debuginfoSrc = "/usr/src/debug"
+type dbgConfig struct {
+	banned bool
+	lib    string
+	src    string
+}
+
+type config struct {
+	emptyDir  string
+	imageBase string
+	outputDir string
+	debuginfo dbgConfig
+}
 
-func readServerINI(path string) error {
+// StateDir is the directory under which swupd will create the update
+// defaults to /var/lib/swupd unless overridden
+var StateDir = "/var/lib/swupd"
+var defaultConfig config
+
+func setDefaultConfig() {
+	defaultConfig = config{
+		emptyDir:  filepath.Join(StateDir, "empty"),
+		imageBase: filepath.Join(StateDir, "image"),
+		outputDir: filepath.Join(StateDir, "www"),
+		debuginfo: dbgConfig{
+			banned: true,
+			lib:    "/usr/lib/debug",
+			src:    "/usr/src/debug",
+		},
+	}
+}
+
+func getConfig() config {
+	setDefaultConfig()
```

Comment:
I've noticed that you changed the defaults to be parametrized based on StateDir -- this wasn't present in the previous code. Can people change StateDir, call getConfig and _expect_ the default values to change?

If no: remove this call and initialize defaultConfig using a struct literal

    var defaultConfig = config{
        imageBase: filepath.Join(...)
    }

If yes: you are essentially passing parameters via a global variable, should avoid that and pass parameters explicitly to the functions that need them. This is fragile. We can fix this later, though.

Reply:
You are right, this is a little messy. I will fix it up.